From 466be17181ba7b78006f982c5fa39a2397ffd748 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Thu, 14 Jun 2018 22:22:03 +0200
Subject: [PATCH] libarchive: add compatibility with python-libarchive >= 2.8

Python librarchive 2.8 removed the mtime_nsec property from
ArchiveEntry so lets wire the ffi function if available
---
 diffoscope/comparators/utils/libarchive.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/diffoscope/comparators/utils/libarchive.py b/diffoscope/comparators/utils/libarchive.py
index f427ede..7bb20c0 100644
--- a/diffoscope/comparators/utils/libarchive.py
+++ b/diffoscope/comparators/utils/libarchive.py
@@ -64,6 +64,10 @@ if not hasattr(libarchive.ffi, 'entry_uname'):
 if not hasattr(libarchive.ffi, 'entry_gname'):
     libarchive.ffi.ffi('entry_gname', [libarchive.ffi.c_archive_entry_p], ctypes.c_char_p)
     libarchive.ArchiveEntry.gname = property(lambda self: libarchive.ffi.entry_gname(self._entry_p))
+# Monkeypatch libarchive-c (<< 2.8)
+if not hasattr(libarchive.ArchiveEntry, 'mtime_nsec') and hasattr(libarchive.ffi, 'entry_mtime_nsec'):
+    libarchive.ArchiveEntry.mtime_nsec = property(
+        lambda self: libarchive.ffi.entry_mtime_nsec(self._entry_p))
 
 # Monkeypatch libarchive-c so we always get pathname as (Unicode) str
 # Otherwise, we'll get sometimes str and sometimes bytes and always pain.
-- 
2.17.1

